### PR TITLE
Add buildConfig enableSigner to skip signing step

### PR DESCRIPTION
**Pull request by joeyleeeeeee97**
All installer related jobs are private so we need to skip these steps when built on other Jenkins. So I propose we use "enableInstallers" to skip the signing step as well.
A failure example in ci.dragonwell : http://ci.dragonwell-jdk.io/view/build_scripts/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-windows-x64-dragonwell/48/

Issue: https://github.com/AdoptOpenJDK/openjdk-build/issues/2231, #2149

### CODE REVIEW

**Comment by M-Davies** [comment]
@sxa or @gdams Can you review this one based off your comments in https://github.com/AdoptOpenJDK/openjdk-build/issues/2149?

**Review by AdamBrousseau — COMMENTED** [review]
It extending the definition of `enableInstallers` to cover signing as well the right answer? I don't know but my initial reaction leans toward no. Is/should the installers and signing be explicitly tied together? Have we documented all the use cases to know if this is this right approach?

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1079]: should throw Exception(e)

**Review by karianna — DISMISSED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (M-Davies) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1079]: There shouldn't be a need for that as the `FlowInterruptedException` is not actually the problem in the timeout situation, it's just signalling that a timeout has been hit

**Comment by joeyleeeeeee97** [comment]
> It extending the definition of `enableInstallers` to cover signing as well the right answer? I don't know but my initial reaction leans toward no. Is/should the installers and signing be explicitly tied together? Have we documented all the use cases to know if this is this right approach?

Yes. I agree 'signing' and 'installing' is not the same thing. I don't know exactly what the signing step does and when are signing and installing serpertated?

 Maybe we need to introduce a new build config here,  like 'ENABLE_SIGNER'.

**Comment by karianna** [comment]
I think we should separate those concerns yes.

**Comment by sxa** [comment]
> @sxa or @gdams Can you review this one based off your comments in #2149?

My comment there was to ask @gdams for input, so I abstain for now and defer to George for additional advice and input

File: pipelines/jobs/pipeline_job_template.groovy
Comment (joeyleeeeeee97) [inline pipelines/jobs/pipeline_job_template.groovy:11]: Please help me verify if it's ok to set false as default and where to activate signer.

**Comment by joeyleeeeeee97** [comment]
Does it seem I broke something during building Openj9?
image

File: pipelines/jobs/pipeline_job_template.groovy
Comment (M-Davies) [reply pipelines/jobs/pipeline_job_template.groovy:11]: This should probably be `true` by default. Like `runInstaller`, `runSigner` should also be placed around Line 19 (inside the `if (binding.hasVariable('PR_BUILDER')) {` block and set to `false`.

**Review by M-Davies — CHANGES_REQUESTED** [review]
Overall looks good! However, as with the [enableInstallers implementation](https://github.com/AdoptOpenJDK/openjdk-build/pull/2015/files), this needs to be added to `config_regeneration.groovy` in order for the pipelines to accept the change. See the linked PR on how to do that. If you need help, there's documentation in [this dir](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/pipelines/build/regeneration) on the regeneration jobs (or just ping me in this PR) 🙂

**Comment by M-Davies** [comment]
> Does it seem I broke something during building Openj9?
> image
> 
> https://github.com/AdoptOpenJDK/openjdk-build/pull/2232/checks?check_run_id=1421903787

Looks like that is an openj9 issue unrelated to this PR. I just ran a [test build](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-linux-x64-openj9/854/console) on the master branch and saw the same error. I guess it's new as I can't see any record of it in the OpenJ9 repository

**Comment by karianna** [comment]
> > Does it seem I broke something during building Openj9?
> > image
> > https://github.com/AdoptOpenJDK/openjdk-build/pull/2232/checks?check_run_id=1421903787
> 
> Looks like that is an openj9 issue unrelated to this PR. I just ran a [test build](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-linux-x64-openj9/854/console) on the master branch and saw the same error. I guess it's new as I can't see any record of it in the OpenJ9 repository -> RAISED [eclipse/openj9#11229](https://github.com/eclipse/openj9/issues/11229)

I don't think you broke anything, a coincidence that there's a nightly bug in OpenJ9 (this happens in Hotspot from time to time as well BTW)

**Comment by joeyleeeeeee97** [comment]
> config_regeneration

Thanks for the advice. Please have another look.

**Comment by M-Davies** [comment]
Code and GitHub tests look good. run tests

**Comment by M-Davies** [comment]
run tests (needed a script approval for the last one)

File: pipelines/jobs/pipeline_job_template.groovy
Comment (M-Davies) [inline pipelines/jobs/pipeline_job_template.groovy:11]: `runSinger` -> `runSigner` (throughout all of this file)

File: pipelines/build/common/build_base_file.groovy
Comment (M-Davies) [inline pipelines/build/common/build_base_file.groovy:145]: `ENABLE_SINGER` -> `ENABLE_SIGNER`

**Review by M-Davies — CHANGES_REQUESTED** [review]
Tester picked up a couple of typo's

File: pipelines/build/common/build_base_file.groovy
Comment (joeyleeeeeee97) [reply pipelines/build/common/build_base_file.groovy:145]: Yes. Could you help me trigger tests again and see how it goes? Thanks for pointing out the typo.

Comment (M-Davies) [reply pipelines/build/common/build_base_file.groovy:145]: Once the code has been updated, ping me and I'll trigger them for you 🙂

Comment (joeyleeeeeee97) [reply pipelines/build/common/build_base_file.groovy]: @M-Davies  Updated, please have another look. Thanks.

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
Script approved. run tests

File: pipelines/build/common/build_base_file.groovy
Comment (M-Davies) [inline pipelines/build/common/build_base_file.groovy:145]:
```suggestion
                ENABLE_SIGNER: enableSigner,
```

File: pipelines/build/common/config_regeneration.groovy
Comment (M-Davies) [inline pipelines/build/common/config_regeneration.groovy:294]:
```suggestion
                ENABLE_SIGNER: true,
```

**Review by M-Davies — CHANGES_REQUESTED** [review]
Couple more typo's I missed (my bad 😓 )

File: pipelines/build/common/config_regeneration.groovy
Comment (joeyleeeeeee97) [reply pipelines/build/common/config_regeneration.groovy]: Sorry for so many typos.
I added pr-tester in our environment now I am trying to figure out how to verify these locally,  next time I would verify locally to be more efficient.

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Review by M-Davies — APPROVED** [review]
Looks good! Thanks for this! 👍

**Review by karianna — APPROVED** [review]
lgtm

**Pull request merged**
